test/test_video_player.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+void setupMockVideoPlayerPlatform() {
+  VideoPlayerPlatform.instance = MockVideoPlayerPlatform();
+}
+
+class MockVideoPlayerPlatform extends Fake
+    with MockPlatformInterfaceMixin
+    implements VideoPlayerPlatform {
+  static const int _textureId = 0xffffffff;
+
+  static StreamController<VideoEvent> _streamController = StreamController<VideoEvent>();
+  static bool initialized = false;
+  static bool isPlaying = false;
+
+  static void reset() {
+    _streamController.close();
+    _streamController = StreamController<VideoEvent>();
+    initialized = false;
+    isPlaying = false;
+  }
+
+  @override
+  Future<void> init() async {}
+
+  @override
+  Future<void> dispose(int textureId) async {
+    assert(textureId == _textureId);
+    initialized = false;
+  }
+
+  @override
+  Future<int?> create(DataSource dataSource) async  {
+    initialized = true;
+    _streamController.add(VideoEvent(
+      eventType: VideoEventType.initialized,
+      duration: const Duration(seconds: 1),
+      size: const Size(0, 0),
+      rotationCorrection: 0,
+    ));
+    return _textureId;
+  }
+
+  @override
+  Stream<VideoEvent> videoEventsFor(int textureId) {
+    assert(textureId == _textureId);
+    return _streamController.stream;
+  }
+
+  @override
+  Future<void> setLooping(int textureId, bool looping) async {
+    assert(textureId == _textureId);
+  }
+
+  @override
+  Future<void> play(int textureId) async {
+    assert(textureId == _textureId);
```

Comment:
These asserts are very helpful! They help significantly reduce the reader's worry that perhaps these methods end up getting called in a way that breaks their assumptions.

Let's add a few more assertions in that spirit:
* `create` check not already initialized
* `dispose` check it is initialized
* couple of other methods add this `textureId` check, where they don't have it already

and I'll add others inline.